src/OrleansAzureUtils/Storage/AzureQueueDataManager.cs

```diff
@@ -140,12 +139,13 @@ public AzureQueueDataManager(string queueName, string deploymentId, string stora
             {
                 // that way we don't have first to create the queue to be able later to delete it.
                 CloudQueue queueRef = queue ?? queueOperationsClient.GetQueueReference(QueueName);
-                var exists = Task<bool>.Factory.FromAsync(queueRef.BeginExists, queueRef.EndExists, null);
+                var exists = queueRef.ExistsAsync();
                 if (await exists)
                 {
-                    await Task.Factory.FromAsync(queueRef.BeginDelete, queueRef.EndDelete, null);
+                    await queueRef.DeleteIfExistsAsync();
```

Comment:
good catch. I tried to make sure that every BeginXXX method was converted to XXXAsync exactly, but I missed this one. I agree this should just use DeleteIfExistsAsync only (and only log if it actually deleted something).